### PR TITLE
AppLayerResult in SMTP

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: https://redmine.openinfosecfoundation.org/issues/4907

### CODE REVIEW

**Comment by suricata-qa** [comment]
ERROR: ASAN TEST FAIL in tlpr1_asan_suri QA test

ERROR: QA failed on tlpr1_asan_suri.
  

Pipeline 5537

File: src/app-layer-smtp.c [hunk 1]
```diff
@@ -566,220 +566,34 @@ int SMTPProcessDataChunk(const uint8_t *chunk, uint32_t len,
...
-            state->input_len = 0;
-
-            return -1;
-
+        uint32_t o_consumed = state->consumed;
+        state->consumed = lf_idx - state->input + 1;
+        state->current_line_len = state->consumed - o_consumed;
+        state->current_line = state->input + o_consumed;
+        state->input_len -= state->current_line_len;
+        if (state->input[state->consumed - 2] == 0x0D) {
```
Comment (victorjulien) [inline src/app-layer-smtp.c:588]: can we be sure here that:
1. `consumed >= 2`
2. `state->input` is big enough in all cases?

File: src/app-layer-smtp.c [hunk 2]
```diff
@@ -566,220 +566,34 @@ int SMTPProcessDataChunk(const uint8_t *chunk, uint32_t len,
...
-            state->input_len -= (lf_idx - state->input) + 1;
-            state->input = (lf_idx + 1);
-
-            return 0;
-        }
-
-        /* toclient */
+    if (lf_idx == NULL) {
+        SCReturnStruct(APP_LAYER_INCOMPLETE(
+                state->consumed, state->input_len));
```
Comment (victorjulien) [inline src/app-layer-smtp.c:581]: this was supposed to be `state->input_len + 1`?

**Comment by inashivb** [comment]
Replaced w #6760

**Pull request closed without merging**